integrations/huggingface-transformers/tutorials/question-answering/question_answering-cli.md

```diff
@@ -0,0 +1,372 @@
+
+### Sparse Transfer Learning with a Custom Teacher
+
+To support the transfer learning process, we can apply model distillation, just like we did for the SQuAD case.
+You are free to use the native Hugging Face workflows to train the dense teacher model (and can even
+pass a Hugging Face model identifier to the command), but you can also use the SparseML CLI as well. 
+
+#### Train The Dense Teacher
+
+Run the following to train a dense teacher model on SquadShifts:
+```
+sparseml.transformers.train.text_classification \
+  --output_dir dense-teacher_rotten_tomatoes \
+  --model_name_or_path zoo:nlp/masked_language_modeling/bert-base/pytorch/huggingface/wikipedia_bookcorpus/base-none \
+  --recipe zoo:nlp/sentiment_analysis/bert-base/pytorch/huggingface/sst2/base-none \
+  --recipe_args '{"init_lr":0.00003}' \
+  --dataset_name rotten_tomatoes --input_column_names "text" --label_column_name "label" \
+  --max_seq_length 128 --per_device_train_batch_size 32 --per_device_eval_batch_size 32 --preprocessing_num_workers 6 \
+  --do_train --do_eval --evaluation_strategy epoch --fp16  \
+  --save_strategy epoch --save_total_limit 1
+```
+
+The model converges to ~72% accuracy without any hyperparameter search.
+
+Note that used the dense version of BERT (the stub ends in `base-none`) as the starting point for the training 
+and passed a recipe from [SparseZoo](https://sparsezoo.neuralmagic.com/models/nlp%2Fsentiment_analysis%2Fbert-base%2Fpytorch%2Fhuggingface%2Fsst2%2Fbase-none) which was used to train the 
+dense teacher for the SST2 task. Since the SST2 task is similiar to the Rotten Tomatoes task, these parameters are a solid
+place to start. This recipe contains no sparsity related modifiers and only controls the learning rate and number of epochs. As such, the script
+will run typical fine-tuning, resulting in a dense model.
+
+Here's what the recipe looks like:
+```yaml
+version: 1.1.0
+
+# General Variables
+num_epochs: 8
+init_lr: 0.00015
+
+# Distillation Variables
+temperature: 2.0
+hardness: 1.0
+
+# Modifiers
+modifiers:
+    - !LearningRateFunctionModifier
+        start_epoch: 0
+        end_epoch: eval(num_epochs)
+        lr_func: linear
+        init_lr: eval(init_lr)
+        final_lr: 0.0
+
+    - !EpochRangeModifier
+        end_epoch: eval(num_epochs)
+        start_epoch: 0.0
+```
+
+While you are free to download and modify the recipe manually (and then pass to SparseML as a local file), you
+can also use `--recipe_args` to modify the recipe on the fly.
+
+In this case, we passed `--recipe_args '{"init_lr": 0.0005, "num_epochs":9}'`. This updates the recipe to run
+for 9 epochs instead of 8 and to use an initial learning rate of `0.0005` instead of `0.00015`.
+
+After running for 9 epochs, the model converges to XX% accuracy.
+
+#### Sparse Transfer Learning with a Custom Teacher
+
+With the dense teacher trained, we can sparse transfer learn with the help of the teacher by passing
```

Comment:
With the dense teacher trained, we can sparse transfer ~~learn~~ with the help of the teacher by passing